File: Server/src/main/java/core/game/content/dialogue/BernaldDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Represents the dialogue plugin used for the bernald npc.
 * @author 'Vexia
 * @version 1.1
 */
@Initializable
public final class BernaldDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code BernaldDialogue} {@code Object}.
	 */
	public BernaldDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code BernaldDialogue} {@code Object}.
	 * @param player the player.
	 */
	public BernaldDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new BernaldDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		player("How are your grapes coming along?");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			npc("Marvellous, thanks to your help, " + player.getUsername() + "!");
			stage = 1;
			break;
		case 1:
			end();
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 2580 };
	}
}

File: Server/src/main/content/region/asgarnia/burthorpe/dialogue/BernaldDialogue.kt
package content.region.asgarnia.burthorpe.dialogue

import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.node.entity.player.Player
import core.plugin.Initializable
import core.tools.END_DIALOGUE
import core.tools.START_DIALOGUE
import org.rs09.consts.NPCs

/**
 * Bernald Dialogue
 * @author ovenbread
 *
 * https://www.youtube.com/watch?v=4odb7qhBPfA
 * https://www.youtube.com/watch?v=nisJKOtqWIo 1:08:53
 */
@Initializable
class BernaldDialogue(player: Player? = null) : DialoguePlugin(player) {

    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
//        // Garden of Tranquility has not been implemented.
//        if(hasRequirement(player!!, "Garden of Tranquility")) {
//            if (isQuestComplete(player!!, DeathPlateau.questName)) {
//                when (stage) {
//                    0 -> playerl(FacialExpression.FRIENDLY, "How are your grapes coming along?").also { stage++ }
//                    1 -> npc(FacialExpression.FRIENDLY, "Marvellous, thanks to your help, " + player.username + "!").also { stage = END_DIALOGUE }
//                }
//            }
//        }
        when (stage) {
            START_DIALOGUE -> npcl(FacialExpression.WORRIED, "Do you know anything about grapevine diseases?").also { stage++ }
            1 -> playerl(FacialExpression.FRIENDLY, "No, I'm afraid I don't.").also { stage++ }
            2 -> npcl(FacialExpression.GUILTY, "Oh, that's a shame. I hope I find someone soon, otherwise I could lose all of this year's crop.").also { stage = END_DIALOGUE }
        }
        return true
    }

    override fun newInstance(player: Player): DialoguePlugin {
        return BernaldDialogue(player)
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.BERNALD_2580)
    }
}
